app/src/main/java/com/github/se/studybuddies/ui/settings/CreateAccount.kt

```diff
@@ -17,59 +17,54 @@ import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.unit.dp
 import com.github.se.studybuddies.navigation.NavigationActions
 import com.github.se.studybuddies.navigation.Route
 import com.github.se.studybuddies.viewModels.UserViewModel
 import com.google.firebase.auth.FirebaseAuth
 import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 @Composable
 fun CreateAccount(userViewModel: UserViewModel, navigationActions: NavigationActions) {
-    val uid = userViewModel.getCurrentUserUID()
+  val uid = userViewModel.getCurrentUserUID()
 
-    val email = FirebaseAuth.getInstance().currentUser?.email ?: ""
-    val usernameState = remember { mutableStateOf("") }
-    val photoState = remember { mutableStateOf(Uri.EMPTY) }
+  val email = FirebaseAuth.getInstance().currentUser?.email ?: ""
+  val usernameState = remember { mutableStateOf("") }
+  val photoState = remember { mutableStateOf(Uri.EMPTY) }
 
-    LaunchedEffect(key1 = true) {
-        val defaultProfilePictureUri = withContext(Dispatchers.IO) { userViewModel.getDefaultProfilePicture() }
-        photoState.value = defaultProfilePictureUri
-    }
+  LaunchedEffect(key1 = true) {
```

Comment:
In my opinion the `key1` argument should not always be equal to `true`. The purpose of this argument is to determine whether or not to cancel and restart the coroutine upon recomposition. So if it is set to `true` it will never be restarted upon recomposition.

[Here](https://developer.android.com/develop/ui/compose/side-effects#restarting-effects) is a most detailed explanation of the `key` argument in LaunchEffect function.

How about giving `uid` as the value of `key1` argument, what do you think ?